src/windows.rs

```diff
@@ -107,3 +110,172 @@ impl PutAddress for ProcessHandle {
         }
     }
 }
+
+/// Use `CreateToolhelp32Snapshot` to get and filter list of loaded modules (called DLLs in Windows)
+/// of this process, returning the base address of it.
+#[allow(clippy::clippy::cast_possible_truncation)] // for size_of as u32
+impl ModuleInfo for Pid {
+    fn get_module_base(&self, name: &str) -> std::io::Result<usize> {
+        // taken from https://stackoverflow.com/questions/41552466/how-do-i-get-the-physical-baseaddress-of-an-dll-used-in-a-process
+        let mut module_entry = tlhelp32::MODULEENTRY32 {
+            dwSize: 0,
+            th32ModuleID: 0,
+            th32ProcessID: 0,
+            GlblcntUsage: 0,
+            ProccntUsage: 0,
+            modBaseAddr: std::ptr::null_mut(), // yikes
+            modBaseSize: 0,
+            hModule: std::ptr::null_mut(), // yikes
+            szModule: [0; tlhelp32::MAX_MODULE_NAME32 + 1],
+            szExePath: [0; winapi::shared::minwindef::MAX_PATH],
+        };
+
+        unsafe {
+            module_entry.dwSize = std::mem::size_of::<tlhelp32::MODULEENTRY32>() as u32;
+
+            let snapshot = tlhelp32::CreateToolhelp32Snapshot(
+                tlhelp32::TH32CS_SNAPMODULE | tlhelp32::TH32CS_SNAPMODULE32,
+                *self,
+            );
+            if snapshot == INVALID_HANDLE_VALUE {
+                return Err(std::io::Error::last_os_error());
+            }
+
+            if tlhelp32::Module32First(snapshot, &mut module_entry) == minwindef::TRUE {
+                // makeshift do-while
+                loop {
+                    if utf8_to_string(&module_entry.szModule) == name {
```

Comment:
Aren't these UTF-16 instead of UTF-8 on Windows?

Reply:
The [Microsoft docs](https://docs.microsoft.com/en-us/windows/win32/api/tlhelp32/ns-tlhelp32-processentry32) have the type defined as `CHAR`. There is also a `PROCESSENTRY32W` equivalent, which uses `WCHAR`. So I assume this is... UTF-8? Or ANSI? I do not actually know, I just took that function from https://github.com/Tommoa/rs-process-memory/blob/master/examples/fastyboy.rs#L22